src/storage/v2/disk/edge_import_mode_cache.cpp

```diff
@@ -0,0 +1,93 @@
+    : in_memory_indices_(Indices(config, StorageMode::IN_MEMORY_TRANSACTIONAL)) {}
+
+InMemoryLabelIndex::Iterable EdgeImportModeCache::Vertices(LabelId label, View view, Transaction *transaction,
+                                                           Constraints *constraints) const {
+  auto *mem_label_index = static_cast<InMemoryLabelIndex *>(in_memory_indices_.label_index_.get());
+  return mem_label_index->Vertices(label, view, transaction, constraints);
+}
+
+InMemoryLabelPropertyIndex::Iterable EdgeImportModeCache::Vertices(
+    LabelId label, PropertyId property, const std::optional<utils::Bound<PropertyValue>> &lower_bound,
+    const std::optional<utils::Bound<PropertyValue>> &upper_bound, View view, Transaction *transaction,
+    Constraints *constraints) const {
+  auto *mem_label_property_index =
+      static_cast<InMemoryLabelPropertyIndex *>(in_memory_indices_.label_property_index_.get());
+  return mem_label_property_index->Vertices(label, property, lower_bound, upper_bound, view, transaction, constraints);
+}
+
+bool EdgeImportModeCache::CreateIndex(LabelId label, PropertyId property,
+                                      const std::optional<ParallelizedIndexCreationInfo> &parallel_exec_info) {
+  auto *mem_label_property_index =
+      static_cast<InMemoryLabelPropertyIndex *>(in_memory_indices_.label_property_index_.get());
+  bool res = mem_label_property_index->CreateIndex(label, property, vertices_.access(), parallel_exec_info);
+  if (res) {
+    scanned_label_properties_.insert({label, property});
+  }
+  return res;
+}
+
+bool EdgeImportModeCache::CreateIndex(LabelId label,
+                                      const std::optional<ParallelizedIndexCreationInfo> &parallel_exec_info) {
+  auto *mem_label_index = static_cast<InMemoryLabelIndex *>(in_memory_indices_.label_index_.get());
+  bool res = mem_label_index->CreateIndex(label, vertices_.access(), parallel_exec_info);
+  if (res) {
+    scanned_labels_.insert(label);
+  }
+  return res;
+}
+
+bool EdgeImportModeCache::VerticesWithLabelPropertyScanned(LabelId label, PropertyId property) const {
+  return VerticesWithLabelScanned(label) || utils::Contains(scanned_label_properties_, std::make_pair(label, property));
+}
+
+bool EdgeImportModeCache::VerticesWithLabelScanned(LabelId label) const {
+  return scanned_all_vertices_ || utils::Contains(scanned_labels_, label);
+}
+
+bool EdgeImportModeCache::AllVerticesScanned() const { return scanned_all_vertices_; }
+
+utils::SkipList<Vertex>::Accessor EdgeImportModeCache::AccessToVertices() { return vertices_.access(); }
+
+utils::SkipList<Edge>::Accessor EdgeImportModeCache::AccessToEdges() { return edges_.access(); }
+
+void EdgeImportModeCache::SetScannedAllVertices() { scanned_all_vertices_ = true; }
+
+utils::Synchronized<std::list<Transaction>, utils::SpinLock> &EdgeImportModeCache::GetCommittedTransactions() {
+  return committed_transactions_;
+}
+
+bool EdgeImportModeCache::DoesNotHaveDeltas() {
```

Comment:
It's a matter of taste, but in my opinion it's better to avoid `Not` in names. Because it could be a mess to understand conditions with this kind of names.

Reply:
method not used so will delete it